include/deal.II/matrix_free/operators.h

```diff
@@ -858,13 +966,40 @@ namespace MatrixFreeOperators
   template <int dim, typename VectorType>
   void
   Base<dim,VectorType>::
-  initialize (std_cxx11::shared_ptr<const MatrixFree<dim,typename Base<dim,VectorType>::value_type> > data_)
+  initialize (std_cxx11::shared_ptr<const MatrixFree<dim,typename Base<dim,VectorType>::value_type> > data_,
+              const std::vector<unsigned int> &given_row_selection,
+              const std::vector<unsigned int> &given_column_selection)
   {
     data = data_;
+
+    selected_rows.clear();
+    selected_columns.clear();
+    if (given_row_selection.empty())
+      for (unsigned int i=0; i<data_->n_components(); ++i)
+        selected_rows.push_back(i);
+    else
+      {
+        for (unsigned int i=0; i<given_row_selection.size(); ++i)
+          {
+            AssertIndexRange(given_row_selection[i], data_->n_components());
+            selected_rows.push_back(given_row_selection[i]);
+          }
+      }
```

Comment:
At the moment we are only checking that the entries of `given_row_selection` are within `[0,data_->n_components())`. Can we also check that they are in fact unique?